azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -48,6 +48,22 @@
 _GET_GOAL_STATE_MAX_ATTEMPTS = 6
 
 
+class GoalStateProperties(object):
+    """
+    Enum for defining the properties that we fetch in the goal state
+    """
+    RoleConfig = 1
+    HostingEnv = 2
+    SharedConfig = 4
+    ExtensionsConfig_Certs = 8
+    RemoteAccessInfo = 16
+
+    @staticmethod
+    def default_properties():
```

Comment:
This is OK, though "default" in the name doesn't quite fit.

An alternative is to add an "All" item to the enum.

Reply:
Updated to add All item